src/test/mir-opt/inline/inline-specialization.rs

```diff
@@ -0,0 +1,48 @@
+#![feature(specialization)]
+
+fn main() {
+    let x = <Vec::<()> as Foo>::bar();
```

Comment:
what exactly is this PR testing? Shouldn't this get inlined? If so, we're probably gonna need the same param env reveal_all change for inlining

Reply:
I think I just wanted to show that the incorrect inlining wasn't happening. However, you're right it ideally would be getting inlined.

Reply:
I've updated the inliner to do the same param env reveal_all change.